### ISSUE TITLE
Recognize file ends with '.cppm', '.ccm', '.c++m', '.cxxm' and '.ixx' as C++ files

### ISSUE THREAD

**Issue by ChuanqiXu9** [github/linguist#6366]
## Describe the enhancement

C++20 Modules is new language feature in C++, which brings new semantics to C++ to allow programmers to import C++ modules. For the module interfaces units, the compilers suggest to  use new extensions to mark them. For example, the clang (https://clang.llvm.org/docs/StandardCPlusPlusModules.html#file-name-requirement) suggests to use '.cppm' (or  '.ccm', '.c++m', '.cxxm') and MSVC suggests to use `.ixx`.

And currently Github won't recognize the files that end with '.cppm', '.ccm', '.c++m', '.cxxm' and '.ixx'  as C++ files. It will be much better if github can recognize them.

**Comment by ChuanqiXu9** [comment]
> * already supported

Oh, sorry for that I misunderstand the current status. And I should make my question clear. Currently when we search codes, it looks like the files ends with .cppm wouldn't be classified into C++. For example, https://github.com/ChuanqiXu9/CMake-Modules-Examples/search?q=export+module

In the languages bar, I could only choose `test`. This confuses me a little bit. Since I want to search how many **C++** projects are using C++20 modules. But it looks like it doesn't work as I expected.

**Comment by lildude** [comment]
Ahh, that's completely out of Linguist's control. That's GitHub's search functionality.

Linguist definitely recognises those files in the repo and they will be included in the righthand sidebar stats:

```console
➜ github-linguist CMake-Modules-Examples/ --breakdown
39.64%  26507      C++
39.64%  26502      C
20.25%  13537      CMake
0.48%   319        Makefile

CMake:
cxx_modules_rules_clang.cmake
two_libraries/CMakeLists.txt
two_static_libraries/CMakeLists.txt
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CMakeASMCompiler.cmake
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CMakeCCompiler.cmake
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CMakeCXXCompiler.cmake
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CMakeSystem.cmake
two_static_libraries/build/cmake_install.cmake

C++:
two_libraries/a.cppm
two_libraries/b.cppm
two_static_libraries/a.cppm
two_static_libraries/b.cppm
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CompilerIdCXX/CMakeCXXCompilerId.cpp
two_static_libraries/main.cpp

C:
two_static_libraries/build/CMakeFiles/3.26.20230220-gd18806e/CompilerIdC/CMakeCCompilerId.c

Makefile:
two_static_libraries/build/CMakeFiles/a.dir/a.cppm.o.ddi.d
two_static_libraries/build/CMakeFiles/b.dir/b.cppm.o.ddi.d
two_static_libraries/build/CMakeFiles/main.dir/main.cpp.o.ddi.d

➜
```

I've just realised, I was mistaken about when `.cppm` was added to Linguist. It was only added last month in https://github.com/github/linguist/pull/6310 which first shipped in Linguist 7.25.0.

GitHub's search uses [go-enry](https://github.com/go-enry/go-enry) for the language detection. go-enry gets its info from Linguist but tends to trail Linguist. In this case, go-enry is still using data from Linguist 7.23.0.

So you're hitting this from the [troubleshooting doc](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes):

> Note: New languages will not appear in GitHub's search results for some time after the pull request has been merged and the new Linguist release deployed to GitHub.com. This is because GitHub's search uses [go-enry](https://github.com/go-enry/go-enry) for language detection which tends to lag behind Linguist by a few weeks to months. This in turn requires an update to the underlying search code once go-enry is inline with Linguist.

**Comment by ChuanqiXu9** [comment]
Thank you very much! This really helps.
